gooddata-flight-server/gooddata_flight_server/config/config.py

```diff
@@ -0,0 +1,515 @@
+        condition=_validate_non_empty_string,
+        cast=str,
+        messages={
+            "condition": f"{_Settings.HealthcheckHost} must be an IP address or hostname.",
+        },
+    ),
+    Validator(
+        _s(_Settings.HealthcheckPort),
+        default=_DEFAULT_HEALTHCHECK_PORT,
+        condition=_validate_non_negative_number,
+        cast=int,
+        messages={
+            "condition": f"{_Settings.HealthcheckPort} must be a valid port number.",
+        },
+    ),
+    Validator(
+        _s(_Settings.MallocTrimIntervalSec),
+        default=_DEFAULT_MALLOC_TRIM_INTERVAL_SEC,
+        condition=_validate_non_negative_number,
+        messages={
+            "condition": f"{_Settings.MallocTrimIntervalSec} must be a positive number.",
+        },
+    ),
+    Validator(
+        _s(_Settings.LogEventKeyName),
+        default=_DEFAULT_LOG_EVENT_KEY_NAME,
+        condition=_validate_non_empty_string,
+        messages={
+            "condition": f"{_Settings.LogEventKeyName} must be a non-empty string.",
+        },
+    ),
+    Validator(
+        _s(_Settings.LogTraceKeys),
+        condition=_validate_mapping,
+        messages={
+            "condition": f"{_Settings.LogTraceKeys} must be a mapping between 'trace_id', 'span_id' "
+            f"and 'parent_span_id' -> keys that should appear in structured log messages.",
+        },
+    ),
+    Validator(
+        _s(_Settings.OtelExporterType),
+        cast=str,
+        condition=_validate_supported_otel_exporter,
+        messages={
+            "condition": f"{_Settings.OtelExporterType} must be one of {', '.join(_SUPPORTED_EXPORTERS)}.",
+        },
+    ),
+    Validator(
+        _s(_Settings.OtelServiceName),
+        cast=str,
+        condition=_validate_non_empty_string,
+        messages={
+            "condition": f"{_Settings.OtelServiceName} must be a non-empty string.",
+        },
+    ),
+    Validator(
+        _s(_Settings.OtelServiceNamespace),
+        cast=str,
+        condition=_validate_non_empty_string,
+        messages={
+            "condition": f"{_Settings.OtelServiceNamespace} must be a non-empty string.",
+        },
+    ),
+    Validator(
+        _s(_Settings.OtelServiceInstanceId),
+        cast=str,
+        condition=_validate_non_empty_string,
+        messages={
+            "condition": f"{_Settings.OtelServiceInstanceId} must be a non-empty string.",
+        },
+    ),
+]
+
+
+def _read_tls_setting(settings: Dynaconf, setting: str) -> Optional[bytes]:
+    value: str = settings.get(setting)
+    if value is None:
+        return None
+
+    value = value.strip()
+    if value.startswith("@"):
+        with open(value[1:], "rb") as f:
+            return f.read()
+
+    return value.encode("ascii")
+
+
+def _create_server_config(settings: Dynaconf) -> ServerConfig:
+    server_settings = settings.get(_SERVER_SECTION_NAME)
+
+    exporter_type = server_settings.get(_Settings.OtelExporterType)
```

Comment:
server_settings can be None, in that case `server_settings.get(...)` fails

Reply:
fixed